### ISSUE TITLE
network visualiser

### ISSUE THREAD

**Issue by rade** [zettio/weave#442]
A visual representation of the network would be valuable to ops and dev folks alike. It could be laced with stats, such as traffic and uptimes.

There are actually three types of network here:

1. The weave network. Weave peers and the connections between them.
2. The application network. Containers, their MACs, IPs, peer association and communication between them.
3. The DNS network. WeaveDNS servers, the traffic between them and to external servers.

The information should be exposed through APIs, and the visualiser component itself ought to be a separate component. Eventually we may also want to add monitoring/alerts, which would be a separate component itself.

This issue is an umbrella; it needs to be split up into smaller pieces.

#359 is a starting point for APIs. #353 and #430 cover extra metrics.

**Comment by bboreham** [comment]
Very rough and simple prototype using VisJS at https://github.com/bboreham/weave/blob/WebUI/wgraph2.html

(you may need something like https://github.com/bboreham/weave/blob/WebUI/httpproxy.go to get past browser CORS restrictions)

**Comment by bboreham** [comment]
Should we close this as the space has been taken over by https://github.com/weaveworks/weave/scope ?

**Comment by rade** [comment]
no, because scope still has no visualisation for weave itself (as opposed to containers running on a weave network).

**Comment by tomwilkie** [comment]
Could mark as a dupe of https://github.com/weaveworks/scope/issues/720

**Comment by rade** [comment]
> The weave network. Weave peers and the connections between them.

This was done on weaveworks/scope#1938
